CorsixTH/Lua/dialogs/edit_room.lua

```diff
@@ -1086,71 +1114,90 @@ local function checkDoorWalls(x, y, wall, has_swingdoor)
     dx = 1
     dy = 0
   end
-
+  local invalid_tile = 0
   if th:getCell(x, y, wall_num) % 0x100 ~= 0 then
-    return false
+    invalid_tile = 4
   end
 
   -- If it is a swing door there are two more locations to check.
   if has_swingdoor then
-    if th:getCell(x - dx, y - dy, wall_num) % 0x100 ~= 0 or
-        th:getCell(x + dx, y + dy, wall_num) % 0x100 ~= 0 then
-      return false
+    if th:getCell(x - dx, y - dy, wall_num) % 0x100 ~= 0 then
+      invalid_tile = invalid_tile + 2
+    end
+    if th:getCell(x + dx, y + dy, wall_num) % 0x100 ~= 0 then
+      invalid_tile = invalid_tile + 8
     end
   end
-  return true
+  return invalid_tile
```

Comment:
Now that you are returning a number instead of a boolean, fix the ```!return``` line in the function comment.